### ISSUE TITLE
[incident] Alpine 3.20 seccomp issues (arm32v6, arm32v7, i386, riscv64, ppc64le)

### ISSUE THREAD

**Issue by tianon** [docker-library/official-images#16830]
Many of the builds for the recent Alpine 3.20 based images are currently failing due to a variant of https://github.com/docker-library/official-images/issues/16829 (although interestingly mostly on less popular architectures such as 32bit x86/arm, riscv64, and ppc64le).  We are aware of the issues, and are working on a fix. :+1:

**Comment by tianon** [comment]
Got all my ducks in a row for builds of a newer Docker engine across all the architectures we support and the problem persists.  I then recall that BuildKit needs the profile updates too, which _might_ be why we're still seeing it. :upside_down_face: :crossed_fingers:

https://github.com/moby/buildkit/blob/v0.12.5/go.mod#L31 is some semi-random commit between v24 and v25, which does _not_ have any of https://github.com/moby/moby/commit/9e10605e771936c8b24d82bc78a01269ac064fea (v26 original merge), https://github.com/moby/moby/commit/7a075cacf9ed0c08560e7ae014e4e90e8ef7ffed (v25 backport), https://github.com/moby/moby/commit/f11b55ffecee4d5545dd168f58e129405e459daa (v24 backport), _or_ https://github.com/moby/moby/commit/5cc674870c2b00e52acf3259e8701a0ffc8da1f1 (v23 backport) :sob:

The _good_ news is that https://github.com/moby/buildkit/blob/v0.13.2/go.mod#L32 _is_ new enough (v25.0.3 is the version that received the backport).  The bad news is that I refactored the way I'm building `tianon/buildkit` and it's temporarily lost a few architectures as a result, so I need to sort that out. :sob:

However, the build I'm still seeing fail is using the default BuildKit-built-into-dockerd (just raw `docker buildx build`, no `--builder`), so it's also very, very possible (and maybe even likely?) that this is something I _also_ need to chase but is unrelated to still seeing the build fail. :sob:

**Comment by tianon** [comment]
Oh man, we probably need `libseccomp2` version 2.5.5+ also, which is actually newer than Debian Stable/bookworm currently has.

![*heavy sigh*](https://github.com/docker-library/official-images/assets/161631/3368c198-fb16-4346-8763-bb951df41dc4)

**Comment by tianon** [comment]
https://github.com/seccomp/libseccomp/compare/v2.5.4...v2.5.5#diff-b37807d701e0c96a05fa13d461c3f4ca7ca00e64b68358d7e5905d94b60858c2R75

Confirmed :sob:

**Comment by tianon** [comment]
Further confirmed -- started an `infosiftr/moby:i386` container on my host, tried `docker build 'https://github.com/tianon/docker-bash.git#967275700d273c985448c889a4e825008cd1fb97:5.2"'` to reproduce the failure, then installed `libseccomp2` version 2.5.5 from unstable and tried the build again with great success (so at least I can apply a workaround in my Docker-in-Docker images!  still not sure about the BuildKit bits but one bridge at a time).

I also filed a bug against `src:libseccomp` in Debian asking for 2.5.5 (or the syscall updates) in either bookworm or bookworm-backports: https://bugs.debian.org/1071822

**Comment by tianon** [comment]
Ok, in theory, fixes are deployed everywhere now.  I'll check back in next week and hopefully everything will be resolved and all these failing Alpine 3.20 builds will be green. :smile:

**Comment by tianon** [comment]
(Reader: he couldn't resist and he wasn't wrong about BuildKit above, so that's next week's project)

**Comment by LaurentGoderre** [comment]
Still seems to be failing on armv6 and v7

**Comment by tianon** [comment]
Doing some hacks in my personal build system to get more architectures on `tianon/buildkit` again (notably `riscv64`): https://github.com/tianon/bashbrew-tianon-meta/commit/371fd85dc02c353954921ce99bc702047c313cb0#diff-dec510d864246250be6cc8d33d2f9040f88ee65dfa31ee3b0e98b4e4ea1fe291

The other half I'll have to solve is using the _host_ architecture for determining which BuildKit image to use instead of using the _target_ architecture (which is a disparity we've had for a while that mostly doesn't matter but in this case becomes important since I'm not hacking it with fake entries anymore).

**Comment by tianon** [comment]
> The other half I'll have to solve is using the _host_ architecture for determining which BuildKit image to use instead of using the _target_ architecture (which is a disparity we've had for a while that mostly doesn't matter but in this case becomes important since I'm not hacking it with fake entries anymore).

Apparently, I've (fortuitously!) forgotten that I _didn't_ actually implement this to that level, and our `arm32v6` and `arm32v5` builders are already running the `arm32v7` image, so we're actually good to go with just that hack! :tada:

**Comment by tianon** [comment]
In _theory_, https://github.com/docker-library/official-images/pull/16856 will fix this (holding off closing this until actual verification though). :crossed_fingers:

**Comment by yosifkit** [comment]
`ppc64le` was successful!

**Comment by tianon** [comment]
This looks like it's settled enough to be closeable. :+1:

**Comment by tianon** [comment]
> I also filed a bug against `src:libseccomp` in Debian asking for 2.5.5 (or the syscall updates) in either bookworm or bookworm-backports: https://bugs.debian.org/1071822

This weekend's stable update included this gem:

```
libseccomp (2.5.4-1+deb12u1) bookworm; urgency=medium

  * Add support for syscalls up to Linux 6.7. (Closes: #1071822)

 -- Felix Geyer <fgeyer@debian.org>  Sat, 25 May 2024 20:36:52 +0200
```

:partying_face: :heart:
